### PR TITLE
Cypress test for button widget happy path

### PR DESCRIPTION
**Pull request by ajith-k-v**
Fixes: #3694
Screenshot 2022-08-10 at 6 48 47 PM

### CODE REVIEW

**Comment by ajith-k-v** [comment]
Multi-select widget happy path

Screenshot 2022-08-10 at 7 04 56 PM

**Review by emidhun — CHANGES_REQUESTED** [review]
I think we can add verification of widget form preview as well.
Thank you.

**Comment by ajith-k-v** [comment]
@emidhun Added the widget verification from preview. Please review

File: cypress-tests/cypress/constants/texts/button.js [hunk 1]
```diff
@@ -1,47 +1,17 @@
-export const buttonText={
-  widgetName: "Button",
+export const buttonText = {
+  defaultWidgetText: "Button",
   buttonProperties: "Properties",
```
Comment (emidhun) [inline cypress-tests/cypress/constants/texts/button.js:3]: I think we can move commonly used labels to commonTexts.

File: cypress-tests/cypress/support/utils/button.js [hunk 2]
```diff
@@ -1,103 +1,39 @@
...
-export const deleteApp = () => {
-  cy.get(commonSelectors.autoSave, { timeout: 9000 }).should("have.text", commonText.autoSave);
-
-  cy.get(commonSelectors.backButton).click();
-  cy.get(commonSelectors.appCardOptions).first().click();
-  cy.get(commonSelectors.deleteApp).click();
-  cy.get(commonSelectors.confirmButton).click();
+import { openAccordion, openEditorSidebar } from "Support/utils/commonWidget";
 
+export const openButtonStylesEditorSideBar = (widgetText, widgetName) => {
```
Comment (emidhun) [inline cypress-tests/cypress/support/utils/button.js:5]: I think we can use the common methods to open the sidebar by modifying the `data-cy` of the button.

**Review by emidhun — CHANGES_REQUESTED** [review]
PR looks good, I think we can add the changes we discussed.
Please let me know if I can do something.

File: cypress-tests/cypress/e2e/editor/widget/buttonHappyPath.cy.js [hunk 3]
```diff
@@ -1,100 +1,344 @@
...
+
+    cy.renameApp(data.appName);
+
+    openEditorSidebar(buttonText.defaultWidgetName);
+    cy.get(commonWidgetSelector.buttonStylesEditorSideBar).click();
+
+    verifyAndModifyStylePickerFx(
+      buttonText.backgroundColor,
+      buttonText.defaultBackgroundColor,
+      data.colourHex
```
Comment (emidhun) [inline cypress-tests/cypress/e2e/editor/widget/buttonHappyPath.cy.js:119]: I think we can use a different color for each color field.
We can assign different values for `data.colourHex`

**Comment by ajith-k-v** [comment]
@emidhun Added review changes. Please review 

Screenshot 2022-09-01 at 10 50 41 AM

**Review by emidhun — APPROVED** [review]
LGTM

**Pull request merged**